alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/gui/view/SingleRunApp.java

```diff
@@ -0,0 +1,459 @@
+        if (monitors != null) {
+            for (final OutputMonitor<T, P> m : monitors) {
+                if (m != null) {
+                    simulation.schedule(() -> m.initialized(simulation.getEnvironment()));
+                }
+            }
+        }
+    }
+
+    /**
+     * Initializes the key bindings.
+     * <p>
+     * Should be overridden to implement keyboard interaction with the GUI.
+     *
+     * @param scene the scene from which the key strokes will be read
+     * @param listener the listener that will run actions on key strokes
+     */
+    protected void initKeybindings(final Scene scene, final KeyboardActionListener listener) {
+        scene.setOnKeyPressed(e -> {
+            if (Keybinds.Companion.get(ActionFromKey.PLAY_AND_PAUSE).filter(key -> key.equals(e.getCode())).isPresent()) {
+                playPauseMonitor.fireEvent(new ActionEvent(e.getSource(), playPauseMonitor));
+                e.consume();
+                return;
+            }
+            listener.action(new KeyboardTriggerAction(ActionOnKey.PRESSED, e.getCode()), e);
+        });
+        scene.setOnKeyReleased(e -> listener.action(new KeyboardTriggerAction(ActionOnKey.RELEASED, e.getCode()), e));
+    }
+
+    /**
+     * The method parses the given parameters from a key-value map.
+     *
+     * @param params the map of parameters name and respective values
+     * @throws IllegalArgumentException if the value is not valid for the parameter
+     * @see Parameters#getNamed()
+     */
+    private void parseNamedParams(final Map<String, String> params) {
+        params.forEach((key, value) -> {
+            if (USE_EFFECT_GROUPS_FROM_FILE.equals(key)) {
+                try {
+                    effectGroups.addAll(EffectSerializer.effectGroupsFromFile(new File(value)));
+                } catch (final IOException e) {
+                    L.warn(e.getMessage());
+                    effectGroups.clear();
+                }
+            } else {
+                L.warn("Unexpected argument " + PARAMETER_NAME_START + key + PARAMETER_NAME_END + value);
+            }
+        });
+    }
+
+    /**
+     * The method parses the given parameters from a list.
+     *
+     * @param params the list of parameters
+     * @see Parameters#getUnnamed()
+     */
+    private void parseUnnamedParams(final List<String> params) {
+        params.forEach(param -> {
+//            try {
+//                switch (param.startsWith(PARAMETER_NAME_START)
+//                        ? param.substring(PARAMETER_NAME_START.length())
+//                        : param) {
+//                    default:
+//                        L.warn("Unexpected argument " + PARAMETER_NAME_START + param);
+//                }
+//            } catch (final IllegalArgumentException e) {
+//                L.warn("Invalid argument: " + param, e);
+//            }
+            // TODO remove if and use the code commented above for new named parameters
```

Comment:
Can you tackle this?

Reply:
I believe the commented code is supposed to be used for handling "unnamed" parameters in the future if any ever get added, but since there are none at the moment, it does nothing (other than outputting a log warning).

[In the past, it used to do more.](https://github.com/Vuksaa/Alchemist/blame/42d9ed2367957e3eb37bd9cb1eacc9e94b99ae1c/alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/SingleRunApp.java#L193) Should I just remove this function?

Reply:
Is the class still used at all? Don't we have two separate launchers for Swing and FX?

Reply:
The current FXUI launcher ([SingleRunFXUI.kt](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/launch/SingleRunFXUI.kt)) makes use of this class to run JavaFX, I mimicked the way [SingleRunSwingUI](https://github.com/AlchemistSimulator/Alchemist/blob/develop/alchemist/alchemist-swingui/src/main/kotlin/it/unibo/alchemist/launch/SingleRunSwingUI.kt) works.